### ISSUE TITLE
[Regression] raise NoMethodError in 2.103.0 when I'm trying to upload ipa with new AppStoreConnect API

### ISSUE THREAD

**Issue by saitjr** [fastlane/fastlane#13284]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

raise NoMethodError in 2.103.0 when I'm trying to upload ipa with new AppStoreConnect API.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `chdir'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `cd'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'</pre>
</details>

**Comment by saitjr** [comment]
It seems `exit_status` be nil ~

**Comment by janpio** [comment]
@chronweigle What is the connection you see between this (an iTunes Transporter failure) to #13286, which fixes an AppStore/iTunes Connect failure?

**Comment by janpio** [comment]
@saitjr Can you please provide more information that is asked for when creating a new issue? https://github.com/fastlane/fastlane/issues/new?template=BUG_REPORT.md I specifically need more output and the lane that is being executed.

Can you please also run the same command again with `--verbose` added at the end? This should give us more information what iTunes Transporter is actually doing.

**Comment by chronweigle** [comment]
The only connection I made was you'd need to be logged in for uploading to work.

**Comment by janpio** [comment]
Thanks @chronweigle, might very well be the root cause - we really need more output to properly debug this. (I actually hope this was the case... because the alternative means really hard debugging of a terrible part of the codebase 🥇 )

**Comment by zolotyx** [comment]
Having the same issue.

<details>
<summary>
Error
</summary>
```
[15:32:54]: [Transporter Error Output]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
[15:32:54]: Transporter transfer failed.
[15:32:54]:
[15:32:54]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
```
</details>
<details>
<summary>
Trace
</summary>
```
/Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /Users/zolotyx/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:128:in `chdir'
	from /Users/zolotyx/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:128:in `cd'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:24:in `block (2 levels) in parsing_binding'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/bin/fastlane:23:in `load'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/bin/fastlane:23:in `<main>'
```
</details>

**Comment by saitjr** [comment]
This error always shown when upload fail. I'm trying to debug it, wait a minute.

```
[17:22:47]: fastlane detected a Gemfile in the current directory
[17:22:47]: however it seems like you don't use `bundle exec`
[17:22:47]: to launch fastlane faster, please use
[17:22:47]: 
[17:22:47]: $ bundle exec fastlane upload_testflight ipa:myapp.ipa account_type:account # this a custom lane called upload_testflight
[17:22:47]: 
[17:22:47]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[17:22:48]: ----------------------------------------
[17:22:48]: --- Step: Verifying fastlane version ---
[17:22:48]: ----------------------------------------
[17:22:48]: Your fastlane version 2.103.1 matches the minimum requirement of 1.106.1  ✅
[17:22:48]: ------------------------------
[17:22:48]: --- Step: default_platform ---
[17:22:48]: ------------------------------
[17:22:48]: Driving the lane 'ios upload_testflight' 🚀
[17:22:48]: -------------------
[17:22:48]: --- Step: pilot ---
[17:22:48]: -------------------
[17:22:49]: Login to App Store Connect (xxx@xxx.com)
[17:22:52]: Login successful
[17:22:53]: Ready to upload new build to TestFlight (App: xxxx)...
[17:22:55]: Going to upload updated app to App Store Connect
[17:22:55]: This might take a few minutes. Please don't interrupt the script.
[17:24:03]: [Transporter Error Output]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
[17:24:03]: Transporter transfer failed.
[17:24:03]: 
[17:24:03]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
+------------------+-----------------------+
|               Lane Context               |
+------------------+-----------------------+
| DEFAULT_PLATFORM | ios                   |
| PLATFORM_NAME    | ios                   |
| LANE_NAME        | ios upload_testflight |
+------------------+-----------------------+

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 💥   | pilot                      | 74          |
+------+----------------------------+-------------+

[17:24:03]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  [Regression] Build Fails with an Invalid 2 step response Error
    https://github.com/fastlane/fastlane/issues/13281 [closed] 31 💬
    10 hours ago

➡️  AGV enabled check fails
    https://github.com/fastlane/fastlane/issues/9506 [closed] 33 💬
    04 Oct 2017

➡️  The App Name you entered has already been used.
    https://github.com/fastlane/fastlane/issues/8079 [closed] 6 💬
    03 May 2017

and 4 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60zero%3F%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `chdir'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `cd'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:41:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by janpio** [comment]
@zolotyx Could you please also run the command with `--verbose` and add all the things the "new issue" template asks for? https://github.com/fastlane/fastlane/issues/new?template=BUG_REPORT.md I need more context here - the stack trace and the error are both just "end results" of something going wrong earlier. 
Also important: What fastlane version are you using?

**Comment by janpio** [comment]
Hey @saitjr, this time this is your actual error:

```
[17:24:03]: [Transporter Error Output]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
[17:24:03]: Transporter transfer failed.
[17:24:03]: 
[17:24:03]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
```

The stacktrace is similar, but your previous issue was probably a different one.

**Comment by saitjr** [comment]
My fault. It should be "itunes_transporter.rb doesn't handle failures during upload very well any more" 🤪

**Comment by janpio** [comment]
Yep, that's the issue where I collect these similar problems. In your last comment though, it is doing pretty much what it should - report and error from iTunes Reporter. The output is just not optimal.

It would be great if we would know what _actual_ error you had when you first opened the issue. Do you still get this crash when you upload a _new_ binary?

**Comment by m17603048828** [comment]
I have the same problem, how to solve it
[Transporter Error Output]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)

**Comment by zedtux** [comment]
That would be great to have some tests on this tool ... this kind of crashes are really annoying. Or you should promote the version for an upgrade when there are issues. I mean I upgraded juste because fastlane was inciting me to do so.
While I really understand and like to be notified about new versions, when a version is known as buggy, it should be promoted, or warning should be displayed or something like that.

**Comment by janpio** [comment]
There are a few thousand tests for fastlane. But as usual with tests, not all complexities can be covered.

Please feel free to open a new issue for your suggestions how fastlane's release process should be improved - this will be overlooked in here and be forgotten as soon as this issue is fixed and gone.

**Comment by zedtux** [comment]
Okay, thank you @janpio for your comment, I checked very quickly (5 seconds) and didn't find any tests.

I'll open another issue as you suggested. 👍

**Comment by joshdholtz** [comment]
This should be fixed now in `2.104.0` 🚀
